sdk/src/types/block/output/delegation.rs

```diff
@@ -360,7 +361,12 @@ impl DelegationOutput {
     }
 
     // Transition, just without full SemanticValidationContext.
-    pub(crate) fn transition_inner(current_state: &Self, next_state: &Self) -> Result<(), StateTransitionError> {
+    pub(crate) fn transition_inner(
+        current_state: &Self,
+        next_state: &Self,
+        context_inputs: &[ContextInput],
+        protocol_parameters: &ProtocolParameters,
```

Comment:
I'm not sure this all should be `transition_inner`. This function should only care about the changes of fields between `current_state` and `next_state`. I would move the new checks to `transition`